src/candid.ts

```diff
@@ -0,0 +1,233 @@
+import { writeFile } from 'fs/promises';
+import { client } from './client';
+import { logger } from './logger';
+import { getToken, oidcOptions } from './oidc';
+import { getBaseFields, getProposals, postPlatformProviderData } from './pdc-api';
+import type { CommandModule } from 'yargs';
+import type { AccessTokenSet } from './oidc';
+
+interface CandidPremierResult {
+  code: number;
+  message: string;
+  data: object;
+}
+
+const getCandidProfile = async (
+  apiKey: string,
+  ein: string,
+): Promise<CandidPremierResult> => {
+  logger.debug(`Looking up EIN ${ein} in Candid Premier API`);
+  const { data } = await client.get<CandidPremierResult>(
```

Comment:
When this returns a 401 with `npm start candid lookup 12-3456789 -- --candid-api-key='testing'`, I don't see the logger boilerplate. It's not a problem, really, but it differs from the behavior exemplified in `npm start error`.

Reply:
Aha, great catch, @bickelj! It turns out error handling behaves differently between synchronously throwing an error and asynchronously returning a Promise that later rejects. I've added a commit to fix that!